src/main/java/uk/ac/ebi/ampt2d/accession/variant/VariantAccessioningRestController.java

```diff
@@ -0,0 +1,52 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.accession.variant;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import uk.ac.ebi.ampt2d.accession.AccessioningService;
+
+import java.util.List;
+import java.util.Map;
+
+@RestController
+@RequestMapping(value = "/v1/accession/variant")
+@ConditionalOnProperty(name = "services", havingValue = "variant-accession")
+public class VariantAccessioningRestController {
```

Comment:
To make this class closer to RESTful principles, it would make sense for both methods to be in the root variant URL `/v1/accession/variant` and then one of the following:

1. `getVariants` would be at least a GET method that receives accession(s) to query, POST if we need many accessions, and `getAccessions` would be a PUT method
2. `getVariants` would support only GET and `getAccessions` would be POST

I would prefer not to encourage people to query thousands of accessions in a single REST request, so maybe option 2 would be my preferred one.

Reply:
will people not query for 100 of samples in a study providing accessions ? i dont know whether the use case apply, correct me if not

Reply:
Maybe, but that is application dependent and it doesn't apply to the variants, where full scans shall be run as a custom batch process.